interface/src/Application.cpp

```diff
@@ -7652,6 +7653,9 @@ void Application::registerScriptEngineWithApplicationServices(const ScriptEngine
     connect(scriptEngine.data(), &ScriptEngine::infoMessage, scriptEngines, &ScriptEngines::onInfoMessage);
     connect(scriptEngine.data(), &ScriptEngine::clearDebugWindow, scriptEngines, &ScriptEngines::onClearDebugWindow);
 
+
+    ScriptEngineLoggingAgent *scriptLogger = new ScriptEngineLoggingAgent(scriptEngine.data());
+    scriptEngine->setAgent(scriptLogger);
```

Comment:
I agree with david that it would be good to disable this by default so it doesn’t affect the performance of normal scripts.  it seems like it adds overhead to every single function call.

I also wonder, is this agent automatically deleted when the script engine dies?  what about the associated thread local storage?

Reply:
Good point, I'll look into it.